test/topology_experimental_raft/test_tablets.py

```diff
@@ -646,3 +646,114 @@ async def check():
+        assert int(tablet_count) == expected_tablet_count
+
+def get_last_tablet_token_for_shard(table_tablets: list[TabletReplicas], host: HostID, shard: int) -> int:
+    for tablet_replica in table_tablets:
+        for host_id, shard_id in tablet_replica.replicas:
+            if host_id == host and shard_id == shard:
+                return tablet_replica.last_token
+    return None
+
+@pytest.mark.asyncio
+async def test_tablet_count_metric_per_shard(manager: ManagerClient):
+    # Given two running servers
+    cmdline = ['--smp=4']
+    config = {'experimental_features': ['consistent-topology-changes', 'tablets']}
+    servers = await manager.servers_add(2, cmdline=cmdline)
+
+    # When two tables are created
+    cql = manager.get_cql()
+    await cql.run_async(f"CREATE KEYSPACE testing WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 1}};")
+    await cql.run_async("CREATE TABLE testing.mytable1 (col1 timestamp, col2 text, col3 blob, PRIMARY KEY (col1));")
+    await cql.run_async("CREATE TABLE testing.mytable2 (col1 timestamp, col2 text, col3 blob, PRIMARY KEY (col1));")
+
+    # Then tablet count metric for each shard is 2
+    expected_count_per_shard = [
+        {"shard_id": 0, "tablet_count": 2},
+        {"shard_id": 1, "tablet_count": 2},
+        {"shard_id": 2, "tablet_count": 2},
+        {"shard_id": 3, "tablet_count": 2},
+    ]
+    await assert_tablet_count_metric_value_for_shards(manager, servers[0], expected_count_per_shard)
+    await assert_tablet_count_metric_value_for_shards(manager, servers[1], expected_count_per_shard)
+
+    # When third table is created
+    await cql.run_async("CREATE TABLE testing.mytable3 (col1 timestamp, col2 text, col3 blob, PRIMARY KEY (col1));")
+
+    # Then tablet count metric for each shard is 3
+    expected_count_per_shard = [
+        {"shard_id": 0, "tablet_count": 3},
+        {"shard_id": 1, "tablet_count": 3},
+        {"shard_id": 2, "tablet_count": 3},
+        {"shard_id": 3, "tablet_count": 3},
+    ]
+    await assert_tablet_count_metric_value_for_shards(manager, servers[0], expected_count_per_shard)
+    await assert_tablet_count_metric_value_for_shards(manager, servers[1], expected_count_per_shard)
+
+    # When one of tables is dropped
+    await cql.run_async("DROP TABLE testing.mytable2;")
+
+    # Then tablet count metric for each shard is 2
+    expected_count_per_shard = [
+        {"shard_id": 0, "tablet_count": 2},
+        {"shard_id": 1, "tablet_count": 2},
+        {"shard_id": 2, "tablet_count": 2},
+        {"shard_id": 3, "tablet_count": 2},
+    ]
+    await assert_tablet_count_metric_value_for_shards(manager, servers[0], expected_count_per_shard)
+    await assert_tablet_count_metric_value_for_shards(manager, servers[1], expected_count_per_shard)
+
+    # When disabling load balancing
+    await manager.api.disable_tablet_balancing(servers[0].ip_addr)
+    await manager.api.disable_tablet_balancing(servers[1].ip_addr)
```

Comment:
It's enough to do it on one server, setting is global.

Reply:
Fixed